### PR TITLE
Alerting: Update logs in notifier package to follow guidelines

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
The package `ngalert.notifier` (except files used by Alertmanager because of refactoring https://github.com/grafana/grafana/pull/58262) is updated to follow the [guidelines ](https://github.com/grafana/grafana/blob/main/contribute/engineering/backend/instrumentation.md):

- all loggers are renamed to start with prefix `ngalert.notifier.`
- all messages updated to start with capital letter
- changed receiver logger name to be `ngalert.notifiers.receivers` + add intergration type (e.g. slack, pagerduty, etc) as context label + added more context like orgID, UID and name of receiver
- changes log messages in receiver to match other loggers
- adds template context to template expansion errors where the template that caused error is clear. This will make messages like `msg="failed to template DingDing message" err="template: :3: undefined variable \"$labels\""` more clear.

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**
This will make alerting logging be compliant with the guidelines, and will improve log reading experience because all loggers will be under the same package

**Which issue(s) does this PR fix?**:
Related to https://github.com/grafana/grafana/issues/56807

### CODE REVIEW

**Review by gotjosh — CHANGES_REQUESTED** [review]
Can we please wait until we merge the Alertmanager replacement, this would give us a bunch of conflicts for code that's not there.

**Comment by yuri-tceretian** [comment]
> Can we please wait until we merge the Alertmanager replacement, this would give us a bunch of conflicts for code that's not there.

I reverted changes in alertmanager files leaving multiorg and receivers changes

**Comment by yuri-tceretian** [comment]
Closing as everything was moved to alerting repo

**Pull request closed without merging**
